packages/demo/backend/src/controllers/wallet.ts

```diff
@@ -185,8 +198,21 @@ export class WalletController {
       const {
         params: { userId },
       } = validation.data
+      const auth = c.get('auth') as AuthContext | undefined
+      if (auth && auth.userId) {
+        const wallet = await walletService.getUserWallet(auth.userId)
+        if (!wallet) {
+          throw new Error('Wallet not found')
+        }
+        const result = await walletService.fundWallet(wallet)
+        return c.json(result)
+      }
 
-      const result = await walletService.fundWallet(userId)
+      const wallet = await walletService.getWallet(userId)
+      if (!wallet) {
+        throw new Error('Wallet not found')
+      }
+      const result = await walletService.fundWallet(wallet)
```

Comment:
Same thoughts [here](https://github.com/ethereum-optimism/verbs/pull/106/files#r2360537386).

Reply:
similar comment here, we should just kill all the non-auth code while building out the new demo. (TODO: https://github.com/ethereum-optimism/verbs/issues/124)